### PR TITLE
feat: initial endpoint for items

### PR DESCRIPTION
**Pull request by davegamboa99**

### CODE REVIEW

File: pages/api/item/add.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import { NextApiRequest, NextApiResponse } from "next";
+import * as z from "zod";
+
+
+const ItemSchema = z.object({
+  name: z.string(),
+  quantity: z.number(),
+  description: z.string(),
+  collectionId: z.number(),
+});
```
Comment (legaspicary) [inline pages/api/item/add.ts:11]: very optional (up to you), to make our project to at least have an effort (not a simple CRUD app):

- can we add a TODO in the future for adding an item edit history

File: pages/api/item/update.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      res
+        .status(200)
+        .json({ category: updateItem, message: "Item successfully updated." });
+    } catch (err) {
+      res.status(500).json({ err, error: "failed to fetch data" });
+    }
+}
+
+function updateIfValueExists(obj: Record<string, any>, key: string, value: any) {
+  return value ? { ...obj, [key]: value } : obj;
```
Comment (legaspicary) [inline pages/api/item/update.ts:65]: could add this in a share lib for utilities (same as `RequestMethod` type)

Comment (legaspicary) [reply pages/api/item/update.ts:65]: we can add a `\lib` folder, what do you think

**Review by legaspicary — APPROVED** [review]
lgtm, some nitpick as always

File: pages/api/item/update.ts (hunk 2 above)
Comment (davegamboa99) [reply pages/api/item/update.ts:65]: > could add this in a share lib for utilities (same as `RequestMethod` type)

yep, already created a task after creating this PR a while ago 
image

File: pages/api/item/add.ts (hunk 1 above)
Comment (davegamboa99) [reply pages/api/item/add.ts:11]: > very optional (up to you), to make our project to at least have an effort (not a simple CRUD app):
> 
> * can we add a TODO in the future for adding an item edit history

sure, we can create a task card for now, and we'll just pull it later on

**Pull request merged**